### PR TITLE
6 get comments by article

### PR DESCRIPTION
**Pull request by Vii182**
Added functionality for new endpoint.

### CODE REVIEW

File: controllers/articles.controller.js [hunk 1]
```diff
@@ -19,5 +19,18 @@ const getArticleById = (req, res, next) => {
...
 
+const getCommentsByArticleId = (req, res, next) => {
+    const { article_id } = req.params;
+    fetchArticleById(article_id).then(() => {
+        return fetchCommentsByArticleId(article_id);
+    })
+    .then((comments) => {
+        res.status(200).send({ comments });
+    }).catch((err) => {
+        console.log(err)
```
Comment (nc-jo-watson) [inline controllers/articles.controller.js:30]: Remember to remove console.logs before doing a PR and merging to main

File: controllers/articles.controller.js [hunk 2]
```diff
@@ -19,5 +19,18 @@ const getArticleById = (req, res, next) => {
     });
 };
 
+const getCommentsByArticleId = (req, res, next) => {
+    const { article_id } = req.params;
+    fetchArticleById(article_id).then(() => {
```
Comment (nc-jo-watson) [inline controllers/articles.controller.js:24]: Nice!

File: __tests__/app.test.js [hunk 3]
```diff
@@ -112,4 +112,41 @@ describe("/api/articles/:article_id", () => {
...
+        })
+    })
+    test("GET: 404 responds with an error when the passed article_id does not exist", () => {
+        return request(app)
+        .get("/api/articles/999/comments")
+        .expect(404)
+        .then(({ body }) => {
+            expect(body.msg).toBe("Article not found");
+        });
+    })
```
Comment (nc-jo-watson) [inline __tests__/app.test.js:151]: You could do with an extra test here to check that when you have an article with no comments an empty array is returned

**Review by nc-jo-watson — COMMENTED** [review]
Really nice code here Matt! 😄 ⭐ Super easy to read and follow your logic, especially your test suite. I like how you've used one of your model functions to check whether the article_id exists, too. The only thing I would say is missing is a test to check what happens when an article with no comments is requested.
